File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] p) {
        int minpri=p[0];
        int maxi=0;
        int cur=0;
        for(int i=1;i<p.length;i++){
            if(minpri>p[i]){
                minpri=p[i];

            }
            cur=p[i]-minpri;
            maxi=Math.max(maxi,cur);
        }
        return maxi;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, p):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxpro=0
        buy=p[0]
        for i in range(len(p)):
            if p[i]<buy:
                buy=p[i]
            if p[i]>buy:
                maxpro=max(maxpro,p[i]-buy)
        return maxpro
   